README.md

```diff
@@ -649,12 +647,405 @@ To begin coding the basic execution functionality of our VM, we only need these
+	var apUpdate ApUpdate
+	var fpUpdate FpUpdate
+
+	if dstRegNum == 1 {
+		dstRegister = FP
+	} else {
+		dstRegister = AP
+	}
+
+	if op0RegNum == 1 {
+		op0Register = FP
+	} else {
+		op0Register = AP
+	}
+
+	switch op1SrcNum {
+        case 0:
+            op1Src = Op1SrcOp0
+        case 1:
+            op1Src = Op1SrcImm
+        case 2:
+            op1Src = Op1SrcFP
+        case 4:
+            op1Src = Op1SrcAP
+        default:
+            return Instruction{}, ErrInvalidOp1RegError
+	}
+
+	switch pcUpdateNum {
+        case 0:
+            pcUpdate = PcUpdateRegular
+        case 1:
+            pcUpdate = PcUpdateJump
+        case 2:
+            pcUpdate = PcUpdateJumpRel
+        case 4:
+            pcUpdate = PcUpdateJnz
+        default:
+            return Instruction{}, ErrInvalidPcUpdateError
+	}
+
+	switch resLogicNum {
+        case 0:
+            if pcUpdate == PcUpdateJnz {
+                res = ResUnconstrained
+            } else {
+                res = ResOp1
+            }
+        case 1:
+            res = ResAdd
+        case 2:
+            res = ResMul
+        default:
+            return Instruction{}, ErrInvalidResError
+	}
+
+	switch opCodeNum {
+        case 0:
+            opcode = NOp
+        case 1:
+            opcode = Call
+        case 2:
+            opcode = Ret
+        case 4:
+            opcode = AssertEq
+        default:
+            return Instruction{}, ErrInvalidOpcodeError
+	}
+
+	switch apUpdateNum {
+        case 0:
+            if opcode == Call {
+                apUpdate = ApUpdateAdd2
+            } else {
+                apUpdate = ApUpdateRegular
+            }
+        case 1:
+            apUpdate = ApUpdateAdd
+        case 2:
+            apUpdate = ApUpdateAdd1
+        default:
+            return Instruction{}, ErrInvalidApUpdateError
+	}
+
+	switch opcode {
+        case Call:
+            fpUpdate = FpUpdateAPPlus2
+        case Ret:
+            fpUpdate = FpUpdateDst
+        default:
+            fpUpdate = FpUpdateRegular
+	}
+    ```
+    
+6. Creating the Instruction and Returning the Result:
+    With all the necessary information extracted and decoded, the method constructs an Instruction object by assigning the decoded values to its fields and returns the created Instruction object along with a nil error if the decoding process is successful.
+
+    ```go 
+    return Instruction{
+		Off0:     offset0,
+		Off1:     offset1,
+		Off2:     offset2,
+		DstReg:   dstRegister,
+		Op0Reg:   op0Register,
+		Op1Addr:  op1Src,
+		ResLogic: res,
+		PcUpdate: pcUpdate,
+		ApUpdate: apUpdate,
+		FpUpdate: fpUpdate,
+		Opcode:   opcode,
+	}, nil
+    ```
+7. Error Handling:
+    If at any point during the decoding process, an unexpected value is encountered or the input doesn't conform to the expected pattern, the method returns an appropriate error. These errors include cases like invalid op1 register, invalid pc update, invalid result logic, invalid opcode, etc.
```

Comment:
```suggestion
	}, nil
    ```

7. Error Handling:
    If at any point during the decoding process, an unexpected value is encountered or the input doesn't conform to the expected pattern, the method returns an appropriate error. These errors include cases like invalid op1 register, invalid pc update, invalid result logic, invalid opcode, etc.
```